pkg/kubelet/status/manager.go

```diff
@@ -382,6 +382,11 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
+			if pod.DeletionGracePeriodSeconds != nil && *pod.DeletionGracePeriodSeconds == 0 {
```

Comment:
@smarterclayton, we still need to restore the initial UPDATE behavior so that kubelet can get the correct grace deletion period for killing containers.

As for minimizing the impact of the race condition, I think it should be rare that kubelet would attempt to delete the object before the apiserver, but it seems to happen in our stressed test environment anyway. The *easy* fix is to change the kubectl test to use `--ignore-not-found` flag for deletion tests. What do you think?